### ISSUE TITLE
FEATURE: Instance-specific hashtags & hashtag moderation

### ISSUE THREAD

**Issue by Pxtl** [mastodon/mastodon#28045]
## Rationale

With the advent of text search, Hashtags are less meaningful now.  But they can still be useful as a discussion topic.  But discussion-websites that get good discussion also give people the ability to *moderate* discussion. But Fedi is instance-based, and hashtags are cross-instance, so nobody can moderate the universe.  So for moderation, we make instance-specific hashtags.  So, the idea is that instance-hashtags would be something in-between a Twitter/Masto hashtag and a Lemmy/Reddit "Community".

## Details

Basically, if the idea is that if I tag a post with `#Ruby@Hachyderm.io` it submits it to Hachyderm's "local" index of `#Ruby`, just as if somebody on Hachyderm.io had posted into `#Ruby`.

If I search for (or _follow_) `#Ruby@Hachyderm.io`, it will search for (or _follow_) all toots to `#Ruby` on Hachyderm.io, even if I'm on another instance.

And the admins of Hachyderm.io can appoint moderators to specific hashtags.  These moderators would have the following abilities on the hashtag:

1) De-index and re-index a specific a toot into/out-of the hashtag.
2) Ban a user from adding their toot to the hashtag index.
3) Same as (2) but an instance, or *all instances*, effectively making the hashtag whitelist-based.
4) Whitelist a user to post to the hashtag.
5) Sticky/De-sticky a toot to the hashtag.

If there is no moderator appointed to a hashtag and no bans exist?  The behavior is as it is today - anybody can post to the hashtag, but with the added wrinkle that people from other (friendly) instances can post to the local hashtag.

### The Implicit Local Hashtag

There would be an implicit unnamed hashtag `#@instancename`.  This hashtag works exactly like the named hashtags except that implicitly all posts will try to use it on their local instance.  This gives users the ability to search and follow the entire feed of another instance by searching or following `#@instancename`.  The "local" pane of your instance could be renamed `#@instancename`, which is basically what it is.  Also, administrators could appoint moderators to `#@instancename`.

### Unscoped hashtags

This is one where we heavily change current behavior: posting to `#myhashtag` without posting an instance-name is posting to your *local* hashtag index.  So if I'm `@Pxtl@Mastodon.social` (which I am), and I toot with a post tagged `#HamOnt`, I'm actually tooting `#HamOnt@Mastodon.social`, exactly like I would be if I tooted at `@gargron` that would actually mean `@gargron@mastodon.social`

Searching (or _following_) `#myhashtag` *without* an instancename would search local indices across all friendly instances, synthesizing the local moderate hashtags of all friendly instances.

### Search/Follow the de-indexed toots of a hashtag

Searching for `##myhashtag` (or `##myhashtag@myinstance.tld`) would search only the posts of a hashtag that have been deindexed.  This allows hashtag-moderators to use `##myhashtag@myinstance.tld` as a moderation queue, and allows others to view `##myhashtag` as a sort of modlog of "what is being hidden from me".

`###myhashtag` is a convenience tool - a union of `#myhashtag` and `##myhashtag`.  The "give me everything" view.

### Motivation

Mastodon instances do not have access to the kind of moderation money that corporate instances have.  Any cases where bottom-up volunteer-based operation can be enabled are ideal.  Reddit/Lemmy have proven the worth of community-based moderation for topic-based communities, going beyond simply creating topical discussion but even producing high-quality locuses of information like /r/AskScience.

I know there are other proposals for "groups", but this approach seems much more "native" to the existing verbs and concepts of Mastodon.

Also, this would provide a clear UI for https://github.com/mastodon/mastodon/issues/6942 which one of the highest-upvoted issues on this repo.

**Comment by Pxtl** [comment]
But that ignores the whole "moderation" side of things.

But also, tag segregation allows for a single UI for segregated tag searching, following, listing, etc.  Easy single-string representation of the act.

And also, what if you want to post to a tag, but you're not on the instance where you agree with the moderation of that tag?

**Comment by ThisIsMissEm** [comment]
I don't think this could be implemented, given the way hashtags work. The current FEP (part of the standards process) [defining hashtags](https://codeberg.org/fediverse/fep/src/branch/main/fep/eb48/fep-eb48.md) is as Mastodon currently has them implemented, which many other fediverse software has copied the implementation of.

To propose a different hashtag system, you'd need to [write a FEP](https://codeberg.org/fediverse/fep/src/branch/main#submitting-a-fep) and get software to adopt it.
